### PR TITLE
[WIP] OpenSSL RSA Bindings

### PR DESCRIPTION
**Pull request by CImrie**
Adds RSA bindings to OpenSSL as mentioned in #3941 

One slight nuance - it can't export to PEM format using a passphrase as OpenSSL fails without any error when performing this. Otherwise works similarly to the [Ruby API](https://ruby-doc.org/stdlib-2.4.0/libdoc/openssl/rdoc/OpenSSL/PKey/RSA.html) but without aliases.

### CODE REVIEW

File: spec/std/openssl/rsa_spec.cr [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+describe OpenSSL::RSA do
+  describe "instantiating and generate a key" do
+    it "can instantiate and generate for a given key size" do
+      pkey = OpenSSL::RSA.new(256)
+      pkey.private?.should be_true
+      pkey.public?.should be_false
+
+      pkey.public_key.public?.should be_true
+    end
+    it "can export to PEM format" do
```
Comment (RX14) [inline spec/std/openssl/rsa_spec.cr]: Typical style is to treat `it` blocks like `def` methods and leave a line between each.

**Comment by CImrie** [comment]
Hey @straight-shoota - At the moment I've pulled out a shard instead so that issues can be ironed out much quicker (independently from crystal stdlib updates) with the intention of it being merged in later as a PR. Does that seem sensible?

**Comment by bararchy** [comment]
@CImrie something as critical as SSL/TLS security should reside in the stdlib

**Comment by RX14** [comment]
but this isn't SSL/TLS.

Eventually `OpenSSL` module will go and be put into a shard as per #5215, and we'll just have a small private openssl binding enough to do TLS and hmac/sha1/etc. Common operations only.

**Comment by straight-shoota** [comment]
Sure. Just wanted to know what's up. @RX14 I'm not sure if it makes sense to merge if it's still work in progress. Could be added later in a new PR either to stdlib or when OpenSSL is extracted to a shard.

**Comment by RX14** [comment]
For now, removing `OpenSSL` is a pipe dream. So i'd rather not refuse PRs that fit into the stdlib *as is* because of something that *might* happen in the future.

**Comment by straight-shoota** [comment]
100% agree. I meant that this contribution is maybe not ready yet. At least that's how I understood @CImrie's statement.

**Comment by RX14** [comment]
Ah I see what you mean now. No, I wasn't proposing to merge this PR, I was proposing to merge a future PR one this happens:

> with the intention of it being merged in later as a PR.

**Comment by bararchy** [comment]
@CImrie what ever happened to this external shard PR ? :) 
Do you still work on it?

**Comment by bararchy** [comment]
@CImrie that's cool to know, I hope we see you in crystal land again soon, I'll go over the project , thanks for your time

**Pull request closed without merging**